Add function to get commenters info

prateek-ghanghas opened the pull request:
# About this PR
1. function "get_commenters_info_for_product_id()" fetches upvoters and commenters info for a product
2. it returns a list of names 

# Testing 
run the code in any python supported code editor. 
provide product id as input
then , it should return a list of names (commenters and upvoters)

prateek-ghanghas committed:
first commit

manmohitgrewal1 commented on functions.py:
```diff
@@ -0,0 +1,60 @@
+
+upvote_list = []
+
+for k in ref:
```
use better variable names. it would be best if you refrained from using names like I, j,e,r,l,  data, ref, etc. 
You should use a variable name such that the reader of the code understand what it hold and it type (like using dict/info/json for dictionary and list/array for lists)

manmohitgrewal1 commented on functions.py:
```diff
@@ -0,0 +1,60 @@
+            for j in range(len(data[i]["items"])):
+                ref.append(data[i]["items"][j]["__ref"])
+
+upvote_list = []
```
good name.Here I know this variable is of type list and will hold stuff related to upvotes

manmohitgrewal1 reviewed (commented):
Looks good to me. Just update the code with requested changes

manmohitgrewal1 commented:
Also, the file name should be `product_hunt_utils.py` because we will have all the utility functions in this file like `get_prev_day_launches()`, `get_list_of_commentors_info()`, etc.


Additionally, can you refactor your code into a function. From my understanding, running this script will get me all the commentors for all the product launched on a particular day. So you can refact the part which will get me commenter info given slug/productID  in the function param. 

Your `product_hunt_utils.py` should look something like this: 

```
import requests
# other imports


def get_commentor_info_for_product_id(product_id):
    """
    Description about the function
    :param product_id: Sample: 1123
    :return: list: Sample:  [...]
    """
    # your code

if __name__ == "__main__":
    get_commentor_info_for_product_id()
` ``

prateek-ghanghas committed:
Update functions.py

variable names modified

manmohitgrewal1 commented:
@prateek-ghanghas  update the PR name from "first commit" to something meaningful and the same goes for the branch name (you can not change the branch name now but keep in mind for next time)

In future if we want to fix anything or look at a specific thing these little things like PR name, PR description, branch name and comments in your code will be very useful.

prateek-ghanghas committed:
Update functions.py

prateek-ghanghas commented:
> Also, the file name should be `product_hunt_utils.py` because we will have all the utility functions in this file like `get_prev_day_launches()`, `get_list_of_commentors_info()`, etc.
> 
> Additionally, can you refactor your code into a function. From my understanding, running this script will get me all the commentors for all the product launched on a particular day. So you can refact the part which will get me commenter info given slug/productID in the function param.

yeah that part was already in the code below

prateek-ghanghas committed:
Update and rename functions.py to product_hunt_utils.py

prateek-ghanghas committed:
Update product_hunt_utils.py

prateek-ghanghas committed:
Update product_hunt_utils.py

About this PR : Added function to fetch commenters and upvoters info when product id is given as input 
Testing : changes in PR are tested by running this code on local machine

manmohitgrewal1 commented on product_hunt_utils.py:
```diff
@@ -0,0 +1,32 @@
+        '''
+        This function fetches commenters and upvoters names by providing the id of that paricular product
+        :param product_id: sample: 402396
+        :return list: sample: [....]
```
this function should return `upvote_and_commenters_list` and you should paste a sample of upvote_and_commenters_list here. This will help us in the future to see how does the output of this function looks like

manmohitgrewal1 commented on product_hunt_utils.py:
```diff
@@ -0,0 +1,32 @@
+                         },
+              'query': 'query PostPageSocialProof($postId:ID!$limit:Int!){post(id:$postId){id contributors(limit:$limit){role user{id ...UserImage __typename}__typename}__typename}}fragment UserImage on User{id name username avatarUrl __typename}',
+                 }
+        res = requests.post('https://www.producthunt.com/frontend/graphql', json=json_data)
```
Make this url as a global variable because we will be using it in different functions. something like this PRODUCT_HUNT_GRAPHQL_API = "https://www.producthunt.com/frontend/graphql"

manmohitgrewal1 commented on product_hunt_utils.py:
```diff
@@ -0,0 +1,32 @@
+def get_commentor_info_for_product_id(product_id):
+
+        '''
+        This function fetches commenters and upvoters names by providing the id of that paricular product
```
good job 👍🏾

prateek-ghanghas committed:
Update product_hunt_utils.py

1. added URL as global variable
2. sample provided for what this function return

manmohitgrewal1 reviewed (approved):
Looks good to me 🚀
